lib/oli/delivery/sections.ex

```diff
@@ -1630,69 +1613,41 @@ defmodule Oli.Delivery.Sections do
       ]
   """
   def get_ordered_container_labels(section_slug, opts \\ []) do
-    short_label = opts[:short_label] || false
-
-    label_for = fn c ->
-      container_label =
-        get_container_label(
-          c.numbering_level,
-          c.customizations || Map.from_struct(CustomLabels.default())
-        )
+    SectionCache.get_or_compute(section_slug, :ordered_container_labels, fn ->
+      fetch_ordered_container_labels(section_slug, opts)
+    end)
+  end
 
-      if short_label do
-        ~s{#{container_label} #{c.numbering_index}}
-      else
-        ~s{#{container_label} #{c.numbering_index}: #{c.title}}
-      end
-    end
+  def fetch_ordered_container_labels(section_slug, opts \\ []) do
+    short_label = opts[:short_label] || false
 
-    ordered_containers =
-      SectionCache.get_or_compute(section_slug, :ordered_containers, fn ->
-        fetch_ordered_containers(section_slug)
-      end)
+    # TODO: OPTIMIZATION replace this with a minimal hierarchy query after v26.2 is merged
+    %Section{customizations: customizations} = get_section_by_slug(section_slug)
+    full_hierarchy = DeliveryResolver.full_hierarchy(section_slug)
```

Comment:
This is not optimal, can be improved once the MinimalHierarchy improvements in v0.26.2 land